lib/src/Uncertainty/Algorithm/Optimization/EfficientGlobalOptimization.cxx

```diff
@@ -302,14 +302,39 @@ void EfficientGlobalOptimization::run()
     }
     solver_.run();
     const OptimizationResult improvementResult(solver_.getResult());
-
     // store improvement
-    Point improvementValue(improvementResult.getOptimalValue());
-    expectedImprovement_.add(improvementValue);
+    Point newPoint(improvementResult.getOptimalPoint());
+    Point newValue;
+    Point improvementValue;
+    if (!problem.hasBounds() || problem.getBounds().contains(newPoint))
+      {
+        improvementValue = improvementResult.getOptimalValue();
+        expectedImprovement_.add(improvementValue);
```

Comment:
This is not the topic of your development, but the naming is as confusing as can be. This is not `expectedImprovement_` but `expectedImprovementHistory_`. And the method `EfficientGlobalOptimization::getExpectedImprovement` should be renamed `EfficientGlobalOptimization::getExpectedImprovementHistory`.

Reply:
The commit has been removed from the PR, the issue being solved in another PR.